exporter/splunkhecexporter/logdata_to_splunk.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecexporter
+
+import (
+	"math"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+const (
+	sourceLabel     = "service.name"
+	sourcetypeLabel = "source.type"
+)
+
+func logDataToSplunk(logger *zap.Logger, ld pdata.Logs, config *Config) ([]*splunkEvent, int) {
+	numDroppedLogs := 0
+	splunkEvents := make([]*splunkEvent, 0)
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		if rl.IsNil() {
+			continue
+		}
+
+		ills := rl.InstrumentationLibraryLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ils := ills.At(j)
+			if ils.IsNil() {
+				continue
+			}
+
+			logs := ils.Logs()
+			for j := 0; j < logs.Len(); j++ {
+				lr := logs.At(j)
+				if lr.IsNil() {
+					numDroppedLogs++
+					continue
+				}
+				ev := mapLogRecordToSplunkEvent(lr, config)
+				splunkEvents = append(splunkEvents, ev)
+			}
+		}
+	}
+
+	return splunkEvents, numDroppedLogs
+}
+
+func mapLogRecordToSplunkEvent(lr pdata.LogRecord, config *Config) *splunkEvent {
+	var host string
+	var source string
+	var sourcetype string
+	fields := map[string]string{}
+	lr.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+		if k == hostnameLabel {
+			host = v.StringVal()
+		} else if k == sourceLabel {
+			source = v.StringVal()
+		} else if k == sourcetypeLabel {
+			sourcetype = v.StringVal()
+		} else {
+			fields[k] = v.StringVal()
```

Comment:
Any intent to handle non-string attributes (i.e. convert them to strings)?

Reply:
Not sure, I assumed `StringVal()` did the conversion for me. Does it fail instead?

Reply:
I believe it just returns empty string if it isn't really a string.  There are no automatic conversions that I know of.

Reply:
Yeah I would either put a comment here about not supporting non-string values or do something to convert non-strings to strings (e.g. for maps/array attribute values you can json encode them).

Reply:
Added an explicit handling stolen from how signalfx does it.